### DISCUSSION TITLE
@poppinss/utils/build/src/Helpers/string.js:241

### DISCUSSION THREAD

**Discussion by aslampatel** [adonisjs/core#3434] [Help]
![Screenshot from 2021-12-19 22-29-16](https://user-images.githubusercontent.com/17266869/146683856-2af26677-de60-4fdb-b180-9d46e06930cc.png)

i have deploy code on server and getting this server i have used node 17 and latest npm also used latest adonis js 5

**Comment by RomainLanz** [comment]
Hey @aslampatel! 👋🏻 

Seems that you are not running the Node version you believe you are running.
Please, double-check that.

**Reply by aslampatel** [reply]
i have run node using Cpanel application manager

**Reply by aslampatel** [reply]
my package.json file 

{
  "name": "junkieauto",
  "version": "1.0.0",
  "private": true,
  "engines": {
    "node": ">=14.0.0"
  },
  "scripts": {
    "dev": "node ace serve --watch",
    "build": "node ace build --production",
    "start": "node server.js"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^5.3.7",
    "@symfony/webpack-encore": "^1.6.1",
    "adonis-preset-ts": "^2.1.0",
    "pino-pretty": "^7.1.0",
    "typescript": "~4.2",
    "youch": "^2.2.2",
    "youch-terminal": "^1.1.1"
  },
  "dependencies": {
    "@adonisjs/auth": "^8.0.10",
    "@adonisjs/core": "^5.4.0",
    "@adonisjs/lucid": "^16.3.2",
    "@adonisjs/mail": "^7.2.4",
    "@adonisjs/repl": "^3.1.7",
    "@adonisjs/session": "^6.1.2",
    "@adonisjs/shield": "^7.0.7",
    "@adonisjs/view": "^6.1.1",
    "fcm-node": "^1.6.1",
    "luxon": "^2.0.2",
    "moment": "^2.29.1",
    "mysql": "^2.18.1",
    "phc-argon2": "^1.1.2",
    "phc-bcrypt": "^1.0.7",
    "proxy-addr": "^2.0.7",
    "reflect-metadata": "^0.1.13",
    "sharp": "^0.29.3",
    "source-map-support": "^0.5.20"
  }
}

**Reply by RomainLanz** [reply]
It does not matter what you put inside the `package.json` file.
Please, run `node -v` inside a terminal to see what version your server has installed.

**Comment by Zeeshan-Haider1214** [comment]
![1111](https://github.com/user-attachments/assets/accb64f6-79b8-4aed-8003-f5f055eb3493)

Locally, I have Node.js version 20, but on cPanel's backend, only Node.js version 10 is running. When I deploy my project, it gives me this error.
please help me how to sol this error?

**Reply by RomainLanz** [reply]
Update your Node.js version.

**Reply by Zeeshan-Haider1214** [reply]
I don't have root level access; how can I update the Node version? Please guide me.

**Reply by Zeeshan-Haider1214** [reply]
Brother please guide me

**Reply by Zeeshan-Haider1214** [reply]
??

**Reply by thetutlage** [reply]
As @RomainLanz said, you need Node.js version >= 21. You need to self figure out how to upgrade it.
